<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TreeDetectorDialogBase</class>
 <widget class="QDialog" name="TreeDetectorDialogBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>500</width>
    <height>500</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Tree Detector</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox_input">
     <property name="title">
      <string>1. Select Inputs</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label_raster">
        <property name="text">
         <string>Input Raster Layer:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QgsMapLayerComboBox" name="mMapLayerComboBox"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_model">
        <property name="text">
         <string>YOLO Model Path:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QgsFileWidget" name="mFileWidget_model"/>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_python">
        <property name="text">
         <string>Processing Python Path:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QgsFileWidget" name="mFileWidget_python"/>
      </item>
      <item row="3" column="1">
       <widget class="QLabel" name="label_python_example">
        <property name="text">
         <string>e.g., /Users/your_name/my_env/bin/python</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_params">
     <property name="title">
      <string>2. Detection Parameters</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="label_confidence">
        <property name="text">
         <string>Confidence Threshold:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QgsDoubleSpinBox" name="mDoubleSpinBox_confidence">
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.050000000000000</double>
        </property>
        <property name="value">
         <double>0.500000000000000</double>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_iou">
        <property name="text">
         <string>IoU Threshold (NMS):</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QgsDoubleSpinBox" name="mDoubleSpinBox_iou">
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.050000000000000</double>
        </property>
        <property name="value">
         <double>0.400000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_detection">
     <property name="title">
      <string>3. Run Detection</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QPushButton" name="btn_start_detection">
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Start Tree Detection</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QProgressBar" name="progressBar">
        <property name="value">
         <number>0</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_status">
        <property name="text">
         <string>Status: Ready</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="button_box">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Close</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsDoubleSpinBox</class>
   <extends>QDoubleSpinBox</extends>
   <header>qgis.gui</header>
  </customwidget>
  <customwidget>
   <class>QgsFileWidget</class>
   <extends>QWidget</extends>
   <header>qgis.gui</header>
  </customwidget>
  <customwidget>
   <class>QgsMapLayerComboBox</class>
   <extends>QComboBox</extends>
   <header>qgis.gui</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>button_box</sender>
   <signal>rejected()</signal>
   <receiver>TreeDetectorDialogBase</receiver>
   <slot>reject()</slot>
  </connection>
 </connections>
</ui>